### ISSUE TITLE
Upload for private model yields "No instances available"

### ISSUE THREAD

**Issue by lewtun** [Kaggle/kagglehub#103]
Hello, thank you for making this utility lib for Kaggle!

I'm trying to upload a local model to Kaggle Hub and want it to be private. However, after following the instructions in the README, I am not able to import the model to a Kaggle notebook and instead see a "No instances available" on the notebooks `Add input` tab:

![Screenshot 2024-04-16 at 21 51 49](https://github.com/Kaggle/kagglehub/assets/26859204/de642395-223c-4b99-9cdf-3ff5dce0f6cf)

Steps to reproduce:

```python
import kagglehub

handle = "lewtun/mistral-7b-sft/pyTorch/v1"
local_files = "./mistral-7b-sft/" # Just a fine-tuned Mistral 7B
kagglehub.model_upload(handle, local_files)
```

It is possible that something is wrong with the variation being set during the upload?  Thanks!

**Comment by lewtun** [comment]
Although my variation doesn't appear on the model page, I notice that if I click on `Add new variation`, then I can see the variation stored under `pyTorch` but as a zipped archive (perhaps this is the problem?)

![Screenshot 2024-04-16 at 21 56 43](https://github.com/Kaggle/kagglehub/assets/26859204/accfeae5-e95d-413b-a184-fa3a63e60e13)

**Comment by rosbo** [comment]
Thanks for reporting. We are looking into this issue. The archive should be decompressed. This is a problem. Stay tuned.

**Comment by lewtun** [comment]
OK it seems the archive does eventually decompress, but for some reason it only includes a subset of the files (e.g. just one of the weights) and seems to have merged the remainder into `special_tokens_map.json` (this file is 5GB when it should be 500 bytes)

![Screenshot 2024-04-19 at 13 18 03](https://github.com/Kaggle/kagglehub/assets/26859204/9a10160d-9f7a-4232-9248-24f6795018e0)

**Comment by CausalTruth** [comment]
I have the same issue. Only one file gets zipped and uploaded, not the complete directory.

**Comment by lewtun** [comment]
These issues now seem to be fixed in `kagglehub==0.2.4`, especially now that files are pushed individually! 

The only caveat that it takes >1h to process the files after upload, but eventually the variation appears on the model page.

**Comment by rosbo** [comment]
Glad to hear to that the issue you faced is now fixed. We are now focused on optimizing the backend processing time for large models.
